src/app/compound/compound.component.ts

```diff
@@ -4,9 +4,9 @@ import {Observable, Subscription} from "rxjs";
 import {CompoundService} from "../service/compound.service";
 import { AuthenticationService } from '../service/authentication.service';
 import {Compound} from "../model/compound";
-import {HttpErrorResponse, HttpResponse} from "@angular/common/http";
-import {MatDialog, MatDialogRef} from "@angular/material/dialog";
-import {ValidationModalComponent} from "./validation-modal/validation-modal.component";
+import { HttpErrorResponse, HttpEvent, HttpResponse, HttpEventType } from "@angular/common/http";
```

Comment:
I was going to use the `HttpEvent` to get the actual progress percent, however it might not be worth it at this point.